### ISSUE TITLE
Abandoned?

### ISSUE THREAD

**Issue by Thorry84** [h4cc/wkhtmltopdf-amd64#43]
This package seems to be abandoned, I would advise people not to use this.

Unfortunately I don't know of a good alternative to this package. A couple of people forked it, updated it and announced they would maintain their version in the future, only to promptly abandon it.

This package isn't required if you have the possibility to get wkhtmltopdf installed natively, which is the preferred solution. But you probably wouldn't be looking here if you have that option available.

If anybody has a real alternative please let me know.

**Comment by bart-v** [comment]
https://github.com/lifewatch/wkhtmltopdf-windows is up to date
Version 0.12.6 is expected soon

**Comment by olsgreen** [comment]
This has been dead for ages.

I'm keeping a fork of this (for linux x64) current here, we're using it in production.

https://github.com/olsgreen/wkhtmltopdf-amd64

**Comment by nicholaszuccarelli** [comment]
@bart-v That package is for Windows only.

**Comment by georgearnall** [comment]
really need an update for 0.12.6 asap

**Comment by olsgreen** [comment]
The guys over at wkhtmltox stopped (for good reason) releasing generic linux binaries after 0.12.4 and started releasing OS specific ones, thus making maintaining a package like this impossible. We target Ubuntu bionic, I maintain a dynamically linked version extracted from the xenial targeted binaries here https://packagist.org/packages/olsgreen/wkhtmltopdf-amd64.

**Comment by georgearnall** [comment]
Fair enough, I've just tried out your package on my bionic evironment and I get the following error:
**libpng12.so.0: cannot open shared object file: No such file or directory**

See more:

```
[2020-06-30 15:24:30] laravel.ERROR: The exit status code '127' says something went wrong:
stderr: "XXX/vendor/olsgreen/wkhtmltopdf-amd64/bin/wkhtmltopdf-amd64: error while loading shared libraries: **libpng12.so.0: cannot open shared object file: No such file or directory**
"
stdout: ""
command: XXX/vendor/olsgreen/wkhtmltopdf-amd64/bin/wkhtmltopdf-amd64 --lowquality --margin-top '50' --footer-html '/tmp/knp_snappy5efb592e245a51.43881784.html' '/tmp/knp_snappy5efb592e244ca6.36246367.html' '/tmp/knp_snappy5efb592e2454e8.31311227.pdf'. {"userId":1,"exception":"[object] (RuntimeException(code: 127): The exit status code '127' says something went wrong:
stderr: \"XXX/vendor/olsgreen/wkhtmltopdf-amd64/bin/wkhtmltopdf-amd64: error while loading shared libraries: libpng12.so.0: cannot open shared object file: No such file or directory
\"
stdout: \"\"
command: XXX/vendor/olsgreen/wkhtmltopdf-amd64/bin/wkhtmltopdf-amd64 --lowquality --margin-top '50' --footer-html '/tmp/knp_snappy5efb592e245a51.43881784.html' '/tmp/knp_snappy5efb592e244ca6.36246367.html' '/tmp/knp_snappy5efb592e2454e8.31311227.pdf'. at XXX/vendor/knplabs/knp-snappy/src/Knp/Snappy/AbstractGenerator.php:378)
```

Any ideas? Do I need to install it ?

**Comment by olsgreen** [comment]
@georgearnall Can you try removing the package `olsgreen/wkhtmltopdf-amd64` and reinstalling? I've just replaced the binary.

**Comment by georgearnall** [comment]
@olsgreen It Works! Thanks for your help this is exactly what I needed.

**Comment by Thorry84** [comment]
If nobody wants to make a generic Linux binary statically linked, this package should be marked as obsolete and other packages shouldn't depend on this. It's great other people want to make replacements for often used platforms such as Ubuntu, but that's far from a drop-in replacement of this package.

Such a shame generic binaries aren't built anymore, but I understand the reasons for it.
